chore: Bump Snaps packages

Mrtenz opened the pull request:
## **Description**

This bumps all Snaps packages to the latest versions and handles breaking changes. It also introduces some new components and RPC methods.

Summary of changes in the snaps deps:
- Add new `snap_dialog` confirmation type that uses fully custom UI
- Add `Container` and `Footer` component
- Add `RadioGroup` component
- Add `Card` component
- Add `snap_getPreferences` to return user preferences
- Add `snap_resolveInterface` to resolve existing interfaces
- Handle initial connections for preinstalled Snaps
- Enforce stricter size limits for source code, static auxiliary files, and `snap_getFile` return values
- Account for revoked origins when calculating the difference in `initialConnections`


Closes https://github.com/MetaMask/MetaMask-planning/issues/1571
Closes https://github.com/MetaMask/MetaMask-planning/issues/2125
Closes https://github.com/MetaMask/snaps/issues/2622
Closes https://github.com/MetaMask/MetaMask-planning/issues/1956

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26086?quickstart=1)

Mrtenz committed:
Bump Snaps packages

GuillaumeRx commented on .yarn/patches/@metamask-snaps-utils-npm-7.7.0-2cc1f044af.patch:
you need to bump `@metamask/snaps-sdk` under `resolutions` (l. 223)

GuillaumeRx replied:
Or remove it ?

GuillaumeRx commented on package.json:
You need to bump `@metamask/snaps-sdk` in the `resolutions` (l.223)

GuillaumeRx replied:
Or remove it ?

Mrtenz replied:
Done. We can't remove it because some packages pull in outdated versions.

Mrtenz committed:
Update resolution for snaps-sdk

Mrtenz commented:
@SocketSecurity ignore npm/@metamask/snaps-utils@7.8.1

This is ours.

FrederikBolding committed:
Update iframe URLs

FrederikBolding committed:
Bump again

FrederikBolding committed:
Handle breaking change with method hooks

FrederikBolding committed:
Update LavaMoat policies

FrederikBolding committed:
Improve typing

FrederikBolding committed:
Add Card component

FrederikBolding committed:
Fix borked imports

FrederikBolding committed:
Handle breaking change for dialog method hooks

FrederikBolding commented:
@SocketSecurity ignore npm/@metamask/snaps-utils@8.0.0

Network access expected.

FrederikBolding committed:
Add missing props to SnapUIImage

david0xd committed:
feat: Add Snap UI Radio component (#26181)

### Waiting on release of: https://github.com/MetaMask/snaps/pull/2592

## **Description**
This PR integrates new Radio components for Snap UI.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26181?quickstart=1)

## **Related issues**
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1571

## **Manual testing steps**
1. Go to `Test Snaps` and install `Interactive UI Snap`.
2. Run Interactive UI Snap by creating a dialog.
3. Check that radio options are available in the UI and confirm their
functionality. (Latest Test Snaps needs to be released before this is
possible, use `main` branch in Snaps repo and run locally if necessary)

## **Screenshots/Recordings**
### **Before**
Nothing to show. Radio options were not available before.

### **After**
![Screenshot 2024-07-29 at 11 29
00](https://github.com/user-attachments/assets/f978ffc5-9c50-4031-a001-4af46f7674fe)
![Screenshot 2024-07-29 at 11 29
17](https://github.com/user-attachments/assets/b8659e75-af9c-455f-a1f1-9fd87b59e35e)
![Screenshot 2024-07-29 at 11 29
40](https://github.com/user-attachments/assets/4d86e1d2-858d-41e8-b383-6c17b711ad1f)

## **Pre-merge author checklist**
- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

FrederikBolding committed:
Hook up snap_getPreferences

FrederikBolding committed:
Update copy and icon for snap_getPreferences

FrederikBolding committed:
Update test-snaps

FrederikBolding committed:
Bump again

FrederikBolding commented:
https://github.com/SocketSecurity ignore npm/@metamask/snaps-utils@8.0.1

Network access expected.

FrederikBolding committed:
Reapply patch

FrederikBolding committed:
Fix casting

GuillaumeRx committed:
feat(snaps): Add custom dialog (#26304)

## **Description**

This PR adds the support for custom dialogs.

Major changes:
- Add support for `Footer` and `Container` custom UI components
- Refactor `snap-ui-renderer` to handle display of the snap header
- Add the new `snap_resolveInterface` snap RPC method
- Add the new `default` dialog type

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26304?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to test-snaps
2. trigger the `custom` snap dialog

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Maarten Zuidhoorn <maarten@zuidhoorn.com>
Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

FrederikBolding committed:
Fix footer styling

Mrtenz commented on app/_locales/en/messages.json:
```diff
@@ -3887,6 +3887,15 @@
+    "message": "Let $1 access information like your preferred language and fiat currency in your MetaMask settings. This helps $1 display content tailored to your preferences. ",
+    "description": "An extended description for the `snap_getPreferences` permission. $1 is the snap name."
+  },
+
```
Seems to be an unnecessary whitespace here.

FrederikBolding replied:
Good catch, fixed!

(curious how the linter doesn't complain about this)

FrederikBolding committed:
Remove newline

Mrtenz reviewed (commented):
Looks good to me. Can't approve my own PR.

FrederikBolding reviewed (approved):
Approving for @Mrtenz - given that I took over this PR

jpuri reviewed (approved):
Changes in files owned by confirmations team look good 👍
